### PR TITLE
Created IBM On Call Manager Alert Transport

### PR DESCRIPTION
**Pull request by SaneiSaya**
Created IBM On Call Manager Alert Transport

I created an alert transport for IBM On Call Manager that will show in the drop down list

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
Hello, Jayna, thanks for your pull request! Looks like your alert transport is based on older LibreNMS code.  You will need to update it to include it upstream.

File: LibreNMS/Alert/Transport/Ibmocm.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    protected $name = 'IBM On Call Manager';
+
+    // Corrected method signature to match the LibreNMS Transport interface
+    public function deliverAlert($alert_data, $opts): bool
+    {
+        $url = $this->config['ocm-url'];
+        $curl = curl_init();
+
+        Proxy::applyToCurl($curl);
+        curl_setopt($curl, CURLOPT_URL, $url);
```
[the remark this answers is not in the document]
Comment (SaneiSaya) [reply LibreNMS/Alert/Transport/Ibmocm.php:29]: I'll get it updated, I'm going through some of the checks now

[the remark this answers is not in the document]
Comment (SaneiSaya) [reply LibreNMS/Alert/Transport/Ibmocm.php]: It's updated now with the new code.

**Comment by SaneiSaya** [comment]
@murrant Do I need to do anything additional before this gets reviewed?

**Comment by laf** [comment]
@SaneiSaya Thanks for submitting this. Code looks good now but just a couple of points:

1. Can you update `doc/Alerting/Transports.md` to add instructions on how to use it.
2. Is their any public documentation for IBM On Call Manager?
3. Does the webhook just accept any arbitrary data sent to it as `$alert_data` is passed across and I'm just wondering how that then gets used.

**Comment by SaneiSaya** [comment]
1. I updated the file
2. There is no public documents I can find, they must be internal only
3. The webhook will accept any data, but with the plugin I made it wi;; send the following to On Call Manager: the name or ip address of the system sending the alert, the name of the alert, the severity, timestamp, the OS, location, and the unique id

**Comment by laf** [comment]
> 1. I updated the file
> 2. There is no public documents I can find, they must be internal only
> 3. The webhook will accept any data, but with the plugin I made it will send the following to On Call Manager: the name or ip address of the system sending the alert, the name of the alert, the severity, timestamp, the OS, location, and the unique id

Thanks, on #3, is that the plugin you've created for IBM on call manager? If so, does that need adding to the IBM system by the user and if so, can you update the docs in this PR to indicate this and how people go about obtaining the plugin.

**Comment by SaneiSaya** [comment]
For number 3 it is the plugin I created for IBM On Call Manager. I added that to the docs already. Do you need additional information, or for me to post information somewhere else?

**Comment by laf** [comment]
I'm wondering how a user of IBM call manager gets the LibreNMS plugin that would allow them to use this transport if it's not documented anywhere?

**Comment by SaneiSaya** [comment]
It's going to be in the pull down list

**Comment by laf** [comment]
> It's going to be in the pull down list

So it's natively supported within IBM call manager?

**Comment by SaneiSaya** [comment]
Yes, I created a plugin for the IBM On Call Manager for LibreNMS as well
and they have approved it already.

File: doc/Alerting/Transports.md [hunk 2]
```diff
@@ -357,6 +357,21 @@ Configuration of the LibreNMS IRC-Bot is described [here](https://github.com/lib
...
+
+API Keys can be found under 'Integrations' in the IBM On Call Manager portal after selecting LibreNMS as the integration.
+
+IBM On Call Manager uses a webhook to send the name of the alert rule. It will include the name or ip address of the system sending the alert, the name of the alert, the severity, timestamp, the OS, location, and the unique id. 
+
+**Example:**
+
+| Config | Example |
+| ------ | ------- |
+| API Key | randomsample |
```
Comment (laf) [inline doc/Alerting/Transports.md:373]: API Key isn't an option in the transport you created.

You also need to document the ocm-url option you have set in the transport.

**Review by laf — CHANGES_REQUESTED** [review]

**Comment by SaneiSaya** [comment]
Requested update added

**Review by laf — APPROVED** [review]
LGTM. Thanks for contributing

**Pull request merged**